reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,81 @@
+                    aws_account_uid = peering.account.uid  # type: ignore[union-attr]
+                    settings = queries.get_secret_reader_settings()
+                    accounts = queries.get_aws_accounts(uid=aws_account_uid)
+                    awsapi = AWSApi(1, accounts, settings=settings, init_users=False)
+                    tags = peering.tags or "{}"  # type: ignore[union-attr]
+                    mesh_results = awsapi.get_vpcs_details(accounts[0], tags)
+                    for mesh_result in mesh_results:
+                        vpc_peering_info = {
+                            "provider": peering.provider,
+                            "vpc_name": mesh_result["vpc_id"],
+                            "cidr_block": mesh_result["cidr_block"],
+                        }
+                        peerings_entries.append(vpc_peering_info)
+                if peering.provider == "account-vpc":
+                    cidr_block = str(peering.vpc.cidr_block)  # type: ignore[union-attr]
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_name": peering.vpc.name,  # type: ignore[union-attr]
+                        "cidr_block": cidr_block,
+                    }
+                    peerings_entries.append(vpc_peering_info)
+                if peering.provider in (
+                    "cluster-vpc-requester",
+                    "cluster-vpc-accepter",
+                ):
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_name": peering.cluster.name,  # type: ignore[union-attr]
+                        "cidr_block": peering.cluster.network.vpc,  # type: ignore[union-attr]
+                    }
+                    peerings_entries.append(vpc_peering_info)
+        overlaps_peering_entries_dict = find_cidr_duplicates_and_overlap(
+            cluster.name, peerings_entries
+        )
+        if not overlaps_peering_entries_dict:
+            return False
+    return True
+
+
+def find_cidr_duplicates_and_overlap(cluster_name: str, input_list: list):
+    for i in range(len(input_list)):  # pylint: disable=consider-using-enumerate
+        compared_vpc = input_list[i]
+        for j in range(i + 1, len(input_list)):
+            comparing_vpc = input_list[j]
+            if ipaddress.ip_network(compared_vpc["cidr_block"]).overlaps(
+                ipaddress.ip_network(comparing_vpc["cidr_block"])
+            ):
+                logging.error(f"VPC peering error in cluster {cluster_name}")
+                logging.error(
+                    f"vpc {compared_vpc['vpc_name']} with cidr block {compared_vpc['cidr_block']} provider by {compared_vpc['provider']} overlaps with vpc {comparing_vpc['vpc_name']} with cidr block {comparing_vpc['cidr_block']} provider by {comparing_vpc['provider']}"
+                )
+                return False
```

Comment:
I mean
```suggestion
def find_cidr_duplicates_and_overlap(cluster_name: str, input_list: list) -> bool:
    for i in range(len(input_list)):  # pylint: disable=consider-using-enumerate
        compared_vpc = input_list[i]
        for j in range(i + 1, len(input_list)):
            comparing_vpc = input_list[j]
            if ipaddress.ip_network(compared_vpc["cidr_block"]).overlaps(
                ipaddress.ip_network(comparing_vpc["cidr_block"])
            ):
                logging.error(f"VPC peering error in cluster {cluster_name}")
                logging.error(
                    f"vpc {compared_vpc['vpc_name']} with cidr block {compared_vpc['cidr_block']} provider by {compared_vpc['provider']} overlaps with vpc {comparing_vpc['vpc_name']} with cidr block {comparing_vpc['cidr_block']} provider by {comparing_vpc['provider']}"
                )
                return True
    return False 
```
We need to return in any case for a function.